tools/SourceKit/tools/sourcekitd/include/sourcekitd/FileSystemProvider.h

```diff
@@ -0,0 +1,32 @@
+//===--- FileSystemProvider.h - ---------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_SOURCEKITD_FILESYSTEMPROVIDER_H
+#define LLVM_SOURCEKITD_FILESYSTEMPROVIDER_H
+
+#include "SourceKit/Support/FileSystemProvider.h"
+#include "llvm/ADT/StringRef.h"
+
+namespace SourceKit {
+
+/// Registers a FileSystemProvider with the running sourcekitd. After this
+/// function is called, requests can use this FileSystemProvider by setting
+/// 'key.vfs.name' to Name.
+///
+/// The caller is responsible for keeping FileSystemProvider alive as long as
+/// sourcekitd.
+void setFileSystemProvider(llvm::StringRef Name,
+                           SourceKit::FileSystemProvider *FileSystemProvider);
```

Comment:
Do you think it would be useful to document this method as thread-unsafe? Perhaps, by changing its name to `setGlobalFileSystemProvider`?